cli/src/commands/git/push.rs

```diff
@@ -946,24 +952,34 @@ fn find_bookmarks_targeted_by_revisions<'a>(
 ) -> Result<Vec<(&'a RefName, LocalAndRemoteRef<'a>)>, CommandError> {
     let mut revision_commit_ids = HashSet::new();
     if use_default_revset {
-        // remote_bookmarks(remote=<remote>)..@
-        let workspace_name = workspace_command.workspace_name();
-        let expression = RevsetExpression::remote_bookmarks(
-            StringPattern::everything(),
-            StringPattern::exact(remote),
-            None,
-        )
-        .range(&RevsetExpression::working_copy(workspace_name.to_owned()))
-        .intersection(&RevsetExpression::bookmarks(StringPattern::everything()));
+        // Get the default revset specified by the user.
+        let push_revset = workspace_command.settings().get_string("revsets.push")?;
+        let aliases_map = RevsetAliasesMap::new();
+        let mut local_variables = HashMap::new();
+        let template_expr = parse_program(&push_revset).map_err(|e| {
+            user_error_with_hint(e, "revsets.push requires to take a `remote` by default")
+        })?;
+        local_variables.insert(remote.as_str(), template_expr);
```

Comment:
Maybe you mean `.insert("remote", remote_expr)`? I think `remote_expr` can be constructed programmatically, but it's also okay to parse formatted `remote` symbol.

Reply:
> Maybe you mean `.insert("remote", remote_expr)`?

what do you mean by `remote_expr` here?

> I think `remote_expr` can be constructed programmatically, but it's also okay to parse formatted `remote` symbol.

Yes, that also can be done, I currently think that if we parse the expression before evaluating it we can at least warn if `revsets.push` doesn't take a remote symbol.

Reply:
> what do you mean by remote_expr here?

The remote name string, something like `ExpressionNode { ExpressionKind::String("origin") }`. We're going to add substitution rule that replaces `remote` symbol/variable to the actual remote name, right?

Reply:
Ok, that finally makes it click for me. I've pushed a version which addresses all of your comments.